common/djangoapps/student/models.py

```diff
@@ -1002,7 +1005,6 @@ def __unicode__(self):
         ).format(self.user, self.course_id, self.created, self.is_active)
 
     @classmethod
-    @transaction.atomic
```

Comment:
@tasawernawaz @schenedx (@cpennington, tagging you as well because I think this might interest you): I took a look at the EnrollmentListView's [post](https://github.com/edx/edx-platform/blob/master/common/djangoapps/enrollment/views.py#L498) method. Requests to this endpoint can be used to create or update enrollments. Unfortunately, in both cases, data is created separately which depends on the enrollment. Examples include updating [mode and activation state](https://github.com/edx/edx-platform/blob/master/common/djangoapps/enrollment/data.py#L233-L236) and updating [enrollment attributes](https://github.com/edx/edx-platform/blob/master/common/djangoapps/enrollment/data.py#L163-L189). If either of these operations fails, we don't want the corresponding enrollments sitting around, orphaned. This means it's not responsible to disable atomic requests for this view, and that the retry logic in this PR won't help.

This isn't an issue that should be addressed as an afterthought, so I'm re-opening ECOM-1730. @tasawernawaz, please use this PR to remove the retry logic introduced in https://github.com/edx/edx-platform/pull/12714 and switch to using `get_or_create`'s `defaults` kwarg, but leave it at that. Doing that will at least remove an unnecessary query, which may help mitigate the IntegrityErrors by shortening transactions involving this method.